### PR TITLE
Update project structure and add Conan recipe

### LINKED ISSUE #2: Project cannot be build by community
Inside `conanfile.txt` we have the following two lines:

```
soxr/0.1.3@audeering/testing
dr_wav/0.12.12@audeering/testing
```

which means the project cannot be build by the community as it fails with

```
soxr/0.1.3@audeering/testing: Not found in local cache, looking in remotes...
soxr/0.1.3@audeering/testing: Trying with 'conancenter'...
ERROR: Unable to find 'soxr/0.1.3@audeering/testing' in remotes
```

compare https://github.com/audeering/audresample/issues/21

### PR DESCRIPTION
**Pull request by Linux13524**
This pull request changes the following things:
- It will cleanup the CMake and C++ project structure a bit
- A Conan recipe is added instead of the `conanfile.txt`
- Dependencies are used from Conan Center, internal references are removed (
- The tests are separated from the main build (`test_package`)
- WAV files are moved into Git LFS

Closes #2

/cc @hagenw

### CODE REVIEW

File: conanfile.py [hunk 1]
```diff
@@ -0,0 +1,46 @@
+from conans import ConanFile, CMake, tools
+
+
+class AudresampleLibConan(ConanFile):
+    name = "audresamplelib"
+    version = "latest"
+    license = "Proprietary"
```
Comment (Linux13524) [inline conanfile.py]: I could not find a license file. Shouldn't we also make this licensed under MIT as audresample?

Comment (hagenw) [reply conanfile.py]: As far as I understood this library, we just use [soxr](https://sourceforge.net/projects/soxr/) as the main ingredient. It is published under LGPLv2. If this is compatible with using MIT here, I would be in favor of switching to MIT.

Comment (chausner-audeering) [reply conanfile.py]: I don't think they are compatible.

Comment (Linux13524) [reply conanfile.py]: Then there would be a license conflict with audresample anyway and we would need to change it.

File: .gitignore [hunk 2]
```diff
@@ -0,0 +1,3 @@
+build
+.vscode
+Release
```
Comment (chausner-audeering) [inline .gitignore:3]: Release?

Comment (Linux13524) [reply .gitignore:3]: Mentioned in the readme.

File: CMakeLists.txt [hunk 3]
```diff
@@ -44,43 +37,20 @@ if(WIN32)
...
-endif()
-
-
-target_include_directories(audresample
-	PRIVATE
-		"${CMAKE_CURRENT_SOURCE_DIR}/include"
-)
+target_include_directories(audresample PRIVATE "${CMAKE_CURRENT_SOURCE_DIR}/include")
 
 find_library(soxr soxr)
```
Comment (chausner-audeering) [inline CMakeLists.txt]:
```suggestion
```
Should be obsolete.

File: test_package/CMakeLists.txt [hunk 4]
```diff
@@ -0,0 +1,36 @@
+cmake_minimum_required(VERSION 3.10.2)
+project(devaice-sdk-base-tests CXX)
```
Comment (chausner-audeering) [inline test_package/CMakeLists.txt]:
```suggestion
project(audresample-tests CXX)
```

File: conanfile.py (hunk 1 above)
Comment (hagenw) [reply conanfile.py]: I think they should not be incompatible. As long as we don't modify the original source code and provide it inside this repo. Most likely we also need to reference that the original code had LGPL. 

All of our files we can specify with MIT license.

Inside `audresample` where we provide only the build libraries we then need to link to the source code of `soxr` inside `audresamplelib`

Comment (hagenw) [reply conanfile.py]: See also from [Wikipedia](https://en.wikipedia.org/wiki/GNU_Lesser_General_Public_License):

> Paragraph 5 of the LGPL version 2.1 states:
>
>A program that contains no derivative of any portion of the Library, but is designed to work with the Library by being compiled or linked with it, is called a "work that uses the Library". Such a work, in isolation, is not a derivative work of the Library, and therefore falls outside the scope of this License.

File: .gitignore (hunk 2 above)
Comment (chausner-audeering) [reply .gitignore:3]: But then build seems unnecessary? How about we change Release to build in the README and get rid of Release here?

**Comment by chausner-audeering** [comment]
Thanks for taking the time to update this to Conan.

File: .gitignore (hunk 2 above)
Comment (Linux13524) [reply .gitignore:3]: Yes, good point. I will update it.

File: conanfile.py (hunk 1 above)
Comment (Linux13524) [reply conanfile.py]: Ah yes. Seems that was the important difference between LGPL and GPL. Where do we need to a reference to the original code?

File: test_package/CMakeLists.txt (hunk 4 above)
Comment (Linux13524) [reply test_package/CMakeLists.txt]: ups ^^

**Comment by Linux13524** [comment]
I also removed the completely outdated and obsolete GitLab-CI config.

File: conanfile.py (hunk 1 above)
Comment (hagenw) [reply conanfile.py]: I think inside https://github.com/audeering/audresample as there we ship only the binary.
To some degree we do this already: 
image

Comment (Linux13524) [reply conanfile.py]: So in this project the link in the README is enough, right? Then I would just also add the MIT here and set it in the conanfile.

Comment (hagenw) [reply conanfile.py]: Yes, sounds sufficient.

Comment (Linux13524) [reply conanfile.py]: Ok, done.

**Comment by Linux13524** [comment]
Can we merge this?

**Comment by chausner-audeering** [comment]
> Can we merge this?

Feel free to merge from my side.

**Comment by hagenw** [comment]
I cannot say something to the content of `test_package/CMakeLists.txt` and `conanfile.py`, but I executed all the steps from the README until build and worked for me, so I guess this seems fine.

**Comment by Linux13524** [comment]
> I cannot say something to the content of `test_package/CMakeLists.txt` and `conanfile.py`, but I executed all the steps from the README until build and worked for me, so I guess this seems fine.

The `test_package` is run when building with `conan create .`

**Pull request merged**